scripts/main/albums.js

```diff
@@ -227,3 +220,30 @@ albums.refresh = function () {
 albums.isTagAlbum = function (albumID) {
 	return albums.json && albums.json.tag_albums.find((tagAlbum) => tagAlbum.id === albumID);
 };
+
+/**
+ * Returns true if the root album is empty in the sense that there is no
+ * visible user content.
+ *
+ * @returns {boolean}
+ */
+albums.isEmpty = function () {
+	return (
+		albums.json === null ||
+		(albums.isSmartAlbumEmpty(albums.json.smart_albums.public) &&
+			albums.isSmartAlbumEmpty(albums.json.smart_albums.recent) &&
+			albums.isSmartAlbumEmpty(albums.json.smart_albums.starred) &&
+			albums.isSmartAlbumEmpty(albums.json.smart_albums.unsorted) &&
+			albums.json.albums.length === 0 &&
+			albums.json.shared_albums.length === 0 &&
+			albums.json.tag_albums.length === 0)
```

Comment:
Some of these seem unnecessary; e.g., public or unsorted smart albums are never defined in public view, and I believe neither is `shared_albums`. But did you perhaps code it this way deliberately so that the method is more generally applicable?

Reply:
Exactly. I know that the albums are never set in anonymous mode, but I wanted to avoid any future surprises for anyone who uses the method under different conditions.